### PR TITLE
Order controller - can add items to cart, checkout, and see confirmation page. Bug with DELETING items from cart. 

### PR DESCRIPTION
**Pull request by AngelaOh**
Did not do any testings yet and taking out items from cart is giving me trouble. Otherwise, would love a second pair of ears and eyes for sound logic and bugs!

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -1,24 +1,70 @@
 class OrdersController < ApplicationController
   def cart # One cart (@order) holds information on zero or many OrderItems
-    @order = Order.new(status: "pending")
-    @items = OrderItem.all
+    @order = Order.find_by(status: "pending") # under the assumption there is only one order happening at a time, does this logic work?
```
Comment (alexandria7) [inline app/controllers/orders_controller.rb:3]: ooh this is interesting. i do wonder if this works. i feel like for our basic implementation it does....do we know that this is only ever going to correspond to one user? (i.e. my brain is thinking, what if multiple people are NOT logged in and using our site and they all add items to their individual carts)

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -1,24 +1,70 @@
...
+    if @order && @order.order_items.length != 0
+      @items = OrderItem.where(order_id: @order.id)
+    end
```
Comment (alexandria7) [inline app/controllers/orders_controller.rb:6]: this logic makes sense to me! :)

File: app/controllers/orders_controller.rb [hunk 3]
```diff
@@ -1,24 +1,70 @@
...
   end
 
   def new_order_item
     # TODO: Quantity should come from product#show view; hardcoded for now
-    @item = OrderItem.new(quantity: 1, order_id: @order.id, product_id: params[:id])
+    if Order.find_by(status: "pending")
+      @order = Order.find_by(status: "pending")
```
Comment (alexandria7) [inline app/controllers/orders_controller.rb:12]: could probably DRY this up but I think this logic makes sense and is great for now!

File: app/controllers/orders_controller.rb [hunk 4]
```diff
@@ -1,24 +1,70 @@
...
+      flash[:success] = "TESTTEST: order has been updated."
+    else
+      @order.errors.messages.each do |field, messages|
+        flash.now[field] = messages
+      end
+      render :new, status: :bad_request
+    end
+
+    redirect_to order_path(params[:id])
+  end
```
Comment (alexandria7) [inline app/controllers/orders_controller.rb:43]: curious about what this method is doing (looks good just looking for clarification). update occurs once the user puts in their payment info?

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -20,12 +20,15 @@
...
 
           <div class="right-nav">
-            <% if session[:user_id] != nil %>
+            <% if session[:user_id] %>
               <li><%= link_to "My Account", user_path(session[:user_id]) %></li>
             <% end %>
             <li><%= link_to "Shopping Cart", cart_path %></li> 
-            <li><%= link_to "Login", github_login_path, class: "btn btn-info" %></li>
-            <li><%= link_to "Logout", logout_path, class: "btn btn-info" , method: :delete %></li>
+            <% if session[:user_id] %>
```
Comment (alexandria7) [inline app/views/layouts/application.html.erb]: should this not be "if !session[:user_id]" or "if session[:user_id] == nil"?

File: app/views/orders/cart.html.erb [hunk 6]
```diff
@@ -1,11 +1,15 @@
...
     <h6>There are no items in your cart right now </h6>
 <% else %>
-    <% @items.each do |item| %>
-        <h6>This is an item with id: <%= item.id %> </h6>
-        <h6>The product name is: <%= Product.find_by(id: item.product_id).name %> </h6>
-
+    <h5>Items in your cart</h5>
+    <p>-------------------------</p>
+    <% @items.each_with_index do |item, i| %>
+        <h6><%= i + 1 %>. <%= link_to Product.find_by(id: item.product_id).name, product_path(item.product_id) %> </h6>
```
Comment (alexandria7) [inline app/views/orders/cart.html.erb:8]: nice Ruby skillz!!

File: app/views/products/index.html.erb [hunk 7]
```diff
@@ -16,11 +17,12 @@
             <%= link_to product_path(product.id) do %>
                 <article class="product-item">
                     <img class="product-page-image" src= <%= product.photo_url %> alt="Image of <%= product.name %>">
-                    <p><%= product.name %></p>
+                    <p><%= link_to product.name, product_path(product.id) %></p>
```
Comment (alexandria7) [inline app/views/products/index.html.erb]: So the way i wrote this makes it so the entire article is clickable! so we don't need to make the product name it's own link (unless you want to! then we should remove the link around the article)

**Review by alexandria7 — APPROVED** [review]
amazing work on orders! happy to look destroy over with you tomorrow

File: app/controllers/orders_controller.rb (hunk 1 above)
Comment (AngelaOh) [reply app/controllers/orders_controller.rb:3]: Omg yes that totally is something that I didn't think about but would affect this logic. Will need to think about how to find_by with something a little more specific than a "pending" status.

File: app/controllers/orders_controller.rb (hunk 4 above)
Comment (AngelaOh) [reply app/controllers/orders_controller.rb:43]: My logic here: once user fills in checkout info and submits, our program should hit orders#update then orders#show so the user sees the confirmation page but the backend program is updating the order instance with update info (name, address, email, etc.) and sending everything to orders#show for confirmation and status change. There may be holes in this logic so please let me know if you see any!!

File: app/views/layouts/application.html.erb (hunk 5 above)
Comment (AngelaOh) [reply app/views/layouts/application.html.erb]: Oopsy, yes absolutely!

File: app/views/products/index.html.erb (hunk 7 above)
Comment (AngelaOh) [reply app/views/products/index.html.erb]: I missed the article being clickable! We should def keep it the way you had it!

**Pull request closed without merging**